packages/core/src/analysis/SubstanceAnalysis.ts

```diff
@@ -236,6 +236,30 @@ export const signatureArgsEqual = (a: Signature, b: Signature): boolean => {
   );
 };
 
+/**
+ * Find all declarations with the same type as a given Identifier in the current program
+ * NOTE: When the statement doesn't exist in the program, `removeStmt` returns the original program without errors.
+ *
+ * @param id a Substance program
+ * @param prog the current program
+ * @returns a new Substance program with the statement removed
+ */
+export const identicalTypeDecls = (
+  ids: Identifier[],
+  prog: SubProg
```

Comment:
Perhaps a more straightforward way is to use `env`: `env.vars` is a symbol table of type `Map<string, TypeConsApp>`. I think you can simply check if the `TypeConsApp`s are equivalent.

Reply:
If I need to pass in Env from `checkSwapInArgs` in `Mutation.ts` into this function, how would I do that?

Reply:
I think the original `Env` is stored in the `SynthesisContext` passed into the guard functions (i.e. `check*`)

Reply:
So it seems that `Identifier` is not directly equatable to `TypeConsApp`. Is there a way to convert between the two/compare the two?

Reply:
I think what I meant was:

* instead of a `SubProg`, you can take in an `Env`. The reason is `env.vars` is a map from `string` ids to their corresponding `TypeConsApp` already, and that seems to be the thing you needed?
* You can find out the types of `ids` by looking up `vars` and find variables of the same type by filtering `vars`, too (if I understood this function correclty).